src/main/java/teammates/logic/core/AccountRequestsLogic.java

```diff
@@ -31,28 +35,108 @@ void initLogicDependencies() {
+        AccountRequestAttributes accountRequestAttributes = accountRequestsDb.createEntity(accountRequest);
+        try {
+            accountRequestAttributes = updateAccountRequest(AccountRequestAttributes
+                    .updateOptionsBuilder(accountRequestAttributes.getEmail(), accountRequestAttributes.getInstitute())
+                    .withStatus(AccountRequestStatus.APPROVED)
+                    .withLastProcessedAt(accountRequestAttributes.getCreatedAt())
+                    .build());
+        } catch (EntityDoesNotExistException ednee) {
+            log.severe("Encountered exception when creating account request: "
+                    + "The newly created account request disappeared before it could be approved.", ednee);
+            throw ednee;
+        }
+        return accountRequestAttributes;
+    }
+
     /**
      * Updates an account request.
      *
      * @return the updated account request
-     * @throws InvalidParametersException if the account request is not valid
+     * @throws InvalidParametersException if the new account request is not valid
      * @throws EntityDoesNotExistException if the account request to update does not exist
+     * @throws EntityAlreadyExistsException if the account request cannot be updated because of an existing account request
      */
     public AccountRequestAttributes updateAccountRequest(AccountRequestAttributes.UpdateOptions updateOptions)
-            throws InvalidParametersException, EntityDoesNotExistException {
+            throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
         return accountRequestsDb.updateAccountRequest(updateOptions);
     }
 
     /**
-     * Creates an account request.
+     * Approves the account request associated with the email address and institute.
      *
-     * @return the created account request
-     * @throws InvalidParametersException if the account request is not valid
-     * @throws EntityAlreadyExistsException if the account request to create already exists
+     * @throws EntityDoesNotExistException if the account request to approve does not exist
      */
-    public AccountRequestAttributes createAccountRequest(AccountRequestAttributes accountRequest)
-            throws InvalidParametersException, EntityAlreadyExistsException {
-        return accountRequestsDb.createEntity(accountRequest);
+    public AccountRequestAttributes approveAccountRequest(String email, String institute)
+            throws EntityDoesNotExistException {
+        try {
+            return updateAccountRequest(AccountRequestAttributes.updateOptionsBuilder(email, institute)
+                    .withStatus(AccountRequestStatus.APPROVED)
+                    .withLastProcessedAt(Instant.now())
+                    .build());
+        } catch (InvalidParametersException | EntityAlreadyExistsException e) {
```

Comment:
The design here follows StudentsLogic#resetStudentGoogleId() even though it may not be appropriate either. I'm not very sure what you mean by "assertion should be done at the layer where this has been validated". Do you think it should pass a boolean variable to the updateAccountRequest method? If so, there are two ways: (1) the boolean variable specifies whether the account request should be validated or not (2) always validate the account request and the boolean variable specifies whether it should throw Assertion exception.

Reply:
Anyway, my opinion why it's written like that is because the method is trying to invoke some service and the service throws some exception that I know shouldn't happen. So that's why I want to throw an Assertion exception. I don't think it's depending on callee to validate the parameters though.

Reply:
Also the two exceptions shouldn't be thrown up because it's not expected to the caller of this approveAccountRequest() method to receive InvalidParametersException or EntityAlreadyExistsException.

Reply:
This is no longer an issue after changing `createAndApproveAccountRequest`.